src/FuncHW/ListExtensions.cs

```diff
@@ -9,7 +9,19 @@ public static class ListExtensions
 
         //add extension for List GetFirst: receives Func, returns the first element in the List that matches the condition
 
-        //add extension for List CountElements: receives Func, returns the number of elements that match the condition
+        public static int CountElements<T>(this List<T> values, Func<T, bool> func)
+        {
+            List<T> resultList = new();
+            foreach (T item in values)
+            {
+                if (func(item))
+                {
+                    resultList.Add(item);
+                }
+            }
+
+            return resultList.Count;
+        }
```

Comment:
i think that creating new list is unnecessary. May be simple counter would be better